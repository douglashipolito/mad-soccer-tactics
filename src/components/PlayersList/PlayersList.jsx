import React from "react";
import { Consumer } from "../Tactics/TacticsContext";
import "./PlayersList.scss";

class PlayersList extends React.Component {
  render() {
    const team = this.props.team;
    if (!team) {
      return null;
    }

    return (
      <Consumer>
        {context => (
          <section
            className="team-container player-list-container"
            key={team.id}
          >
            <h2 className="team-name">{team.name}</h2>
            <div className="player-list">
              <ul className="team-list">
                {team.players.map(player => (
                  <li
                    className={`player${player.selected ? " selected" : ""}`}
                    key={player.id}
                  >
                    <h3 className="player-name">
                      <span>
                        {player.name} - <small>{player.role}</small>
                      </span>
                    </h3>
                    <figure
                      className="player-image-container"
                      onDragStart={event =>
                        !player.selected
                          ? context.handleDragStart(event, team.id, player.id)
                          : false
                      }
                      onDragLeave={event =>
                        context.handleDragLeave(event, team.id)
                      }
                      onDragEnd={event => context.handleDragEnd(event, team.id)}
                      draggable={team.configurable ? !player.selected : false}
                    >
                      <img
                        className="player-avatar"
                        src={player.avatar}
                        alt={player.name}
                      />
                    </figure>
                  </li>
                ))}
              </ul>
            </div>
          </section>
        )}
      </Consumer>
    );
  }
}

export default PlayersList;
