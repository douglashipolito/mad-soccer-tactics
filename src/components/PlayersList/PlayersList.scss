.player-list-container {
  ul {
    list-style: none;
    margin: 0;
    padding: 0;
  }

  .player-image-container {
    width: 100px;
    height: 100px;
    margin: 0 auto;
    background: #f1f1f1;
    border-radius: 50%;
    overflow: hidden;
    box-shadow: 0px 1px 4px 0px #717171;
    border: 2px solid #717171;

    .player-avatar {
      width: 100%;
    }

    &:hover {
      cursor: move;
      box-shadow: 2px 4px 14px 4px #717171;
    }
  }

  .player {
    &.selected {
      .player-image-container {
        opacity: 0.5;

        &:hover {
          cursor: no-drop;
        }
      }
    }
  }
}
