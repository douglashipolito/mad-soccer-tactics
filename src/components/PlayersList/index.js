import PlayersList from "./PlayersList.jsx";

export default PlayersList;
