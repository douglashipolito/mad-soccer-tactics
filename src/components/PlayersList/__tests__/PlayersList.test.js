import React from "react";
import { create } from "react-test-renderer";
import PlayersList from "../";

test("Players List Renders", () => {
  const string = create(<PlayersList />);
  expect(string).toMatchSnapshot();
});
