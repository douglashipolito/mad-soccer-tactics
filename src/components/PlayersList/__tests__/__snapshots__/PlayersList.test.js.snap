// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Players List Renders 1`] = `null`;
